src/qibolab/pulses.py

```diff
@@ -1403,6 +1406,57 @@ def serial(self):
         return f"FluxPulse({self.start}, {self.duration}, {format(self.amplitude, '.6f').rstrip('0').rstrip('.')}, {self.shape}, {self.channel}, {self.qubit})"
 
 
+class CouplerFluxPulse(Pulse):
```

Comment:
Do we really need a new `Pulse` for the couplers? Can't we just use the existing `FluxPulse` by extending the `qubit` argument to be either a `Qubit` or a `Coupler`? What is the physical difference between a `CouplerFluxPulse` and a `FluxPulse`?

Reply:
It would work what you suggest, I just found it simpler or more readable to separete them.

https://github.com/qiboteam/qibolab/pull/575#discussion_r1315677028